homeassistant/components/sensor/canary.py

```diff
@@ -59,27 +71,48 @@ def state(self):
     def unique_id(self):
         """Return the unique ID of this sensor."""
         return "sensor_canary_{}_{}".format(self._device_id,
-                                            self._sensor_type.value)
+                                            self._sensor_type[0])
 
     @property
     def unit_of_measurement(self):
-        """Return the unit of measurement this sensor expresses itself in."""
-        from canary.api import SensorType
-        if self._sensor_type == SensorType.TEMPERATURE:
-            return TEMP_CELSIUS if self._is_celsius else TEMP_FAHRENHEIT
-        elif self._sensor_type == SensorType.HUMIDITY:
-            return "%"
-        elif self._sensor_type == SensorType.AIR_QUALITY:
-            return ""
+        """Return the unit of measurement."""
+        return self._sensor_type[1]
+
+    @property
+    def icon(self):
+        """Icon for the sensor."""
+        return self._sensor_type[2]
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        if self._sensor_type[0] == "air_quality":
+            air_quality = None
+            if self._sensor_value <= .4:
```

Comment:
If sensor value is None this comparison will error. I'd add a check to the if statement above.